File: src/leetcode/2025/12/Solution231.java
public class Solution231 {
    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }

        return (n & (n - 1)) == 0;
    }

    public static void main(String[] args) {
        Solution231 sol = new Solution231();

        int n1 = 1;
        System.out.println(sol.isPowerOfTwo(n1));

        int n2 = 16;
        System.out.println(sol.isPowerOfTwo(n2));

        int n3 = 3;
        System.out.println(sol.isPowerOfTwo(n3));
    }
}

File: src/leetcode/2025/12/Solution231.cpp
#include <iostream>
using namespace std;

class Solution231
{
public:
    bool isPowerOfTwo(int n)
    {
        return n > 0 && (n & (n - 1)) == 0;
    }
};

int main()
{
    Solution231 sol;

    int n1 = 1;
    bool r1 = sol.isPowerOfTwo(n1);
    cout << r1 << endl;

    int n2 = 16;
    bool r2 = sol.isPowerOfTwo(n2);
    cout << r2 << endl;

    int n3 = 3;
    bool r3 = sol.isPowerOfTwo(n3);
    cout << r3 << endl;

    return 0;
}